eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxo.hs

```diff
@@ -27,11 +33,31 @@ instance InjectRuleFailure "UTXO" AlonzoUtxoPredFailure (ConwayEra c) where
 instance InjectRuleFailure "UTXO" ShelleyUtxoPredFailure (ConwayEra c) where
   injectFailure =
     AlonzoInBabbageUtxoPredFailure
-      . allegraToAlonzoUtxoPredFailure
+      . allegraToConwayUtxoPredFailure
       . shelleyToAllegraUtxoPredFailure
 
-instance InjectRuleFailure "UTXO" AllegraUtxoPredFailure (ConwayEra c) where
-  injectFailure = AlonzoInBabbageUtxoPredFailure . allegraToAlonzoUtxoPredFailure
+instance InjectRuleFailure "UTXO" Allegra.AllegraUtxoPredFailure (ConwayEra c) where
+  injectFailure = AlonzoInBabbageUtxoPredFailure . allegraToConwayUtxoPredFailure
 
 instance InjectRuleFailure "UTXO" AlonzoUtxosPredFailure (ConwayEra c) where
   injectFailure = AlonzoInBabbageUtxoPredFailure . UtxosFailure
+
+allegraToConwayUtxoPredFailure ::
+  forall era.
+  EraRuleFailure "PPUP" era ~ VoidEraRule "PPUP" era =>
+  Allegra.AllegraUtxoPredFailure era ->
+  AlonzoUtxoPredFailure era
+allegraToConwayUtxoPredFailure = \case
+  Allegra.BadInputsUTxO x -> BadInputsUTxO x
+  Allegra.OutsideValidityIntervalUTxO vi slotNo -> OutsideValidityIntervalUTxO vi slotNo
+  Allegra.MaxTxSizeUTxO x y -> MaxTxSizeUTxO x y
+  Allegra.InputSetEmptyUTxO -> InputSetEmptyUTxO
+  Allegra.FeeTooSmallUTxO c1 c2 -> FeeTooSmallUTxO c1 c2
+  Allegra.ValueNotConservedUTxO vc vp -> ValueNotConservedUTxO vc vp
+  Allegra.WrongNetwork x y -> WrongNetwork x y
+  Allegra.WrongNetworkWithdrawal x y -> WrongNetworkWithdrawal x y
+  Allegra.OutputTooSmallUTxO x -> OutputTooSmallUTxO x
+  Allegra.UpdateFailure x -> absurdEraRule @"PPUP" @era x
+  Allegra.OutputBootAddrAttrsTooBig xs -> OutputTooBigUTxO (map (0,0,) xs)
```

Comment:
I wonder  why it's not `OutputBootAddrAttrsTooBig xs` here, but I can see that this is how it is in `allegraToAlonzoUtxoPredFailure`, so it's probably correct.

Reply:
That is an interesting observation. I have not looked into why it was that way. I only copy pasted the `allegraToAlonzoUtxoPredFailure` and adjusted the part that was relevant for Conway.

It looks like a mistake in injection that was introduced here: https://github.com/IntersectMBO/cardano-ledger/commit/47d8a35bf31fdd5c1bb5c20b40abd8303884e13f#diff-a9a408a2190f9a7d639380d4faf3d0d0178c9b96d2195c42d8bf7fdd505faa3aR566

I'll fix it for Conway and we can clean it up for older eras later.

Reply:
Done